### PR TITLE
docs(deploy): Adding confluent cloud doc

### PR DESCRIPTION
**Pull request by jjoyce0510**
Title says it. Adding a doc for how to deploy using confluent cloud. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docs/deploy/confluent-cloud.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+springKafkaConfigurationOverrides:
+      security.protocol: SASL
+      sasl.jaas.config: org.apache.kafka.common.security.plain.PlainLoginModule   required username='XFA35EL7QFUQP2PA' password='zmkfP6Envr9TYutsjLB3ZYfrk+yfCXD8sQHCE3EMp57A2jNs4RR7J1bU9k6lM6rU'; # Provided inside Confluent Control Center
+      sasl.mechanism: PLAIN
+      client.dns.lookup: use_all_dns_ips
+      basic.auth.credentials.source: USER_INFO
+      basic.auth.user.info: U4UXEN6SP3KEWMO4:ESju8ZfETDm2RZo/6KO057hPYs2TGjFKmvMWUFnlJ3uKubFbB1Sfs7aOjUVT6CGe # Provided within Confluent Control Center
+```
+
+Typically, we suggest that all secrets are managed using K8s native secret manager. If you are using this, you can replace actual credentials with references to your secrets. For example,
```
Comment (aseembansal-gogo) [inline docs/deploy/confluent-cloud.md:100]: I don't think the way it is given will use K8s native secret manager. What I had given was a ansible template. So these were being replaced by ansible.

Comment (jjoyce0510) [reply docs/deploy/confluent-cloud.md:100]: yes - good catch. updating the doc

File: docs/deploy/confluent-cloud.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+kafkaSetupJob:
+    enabled: false
+```
+
+Then, update the `kafka` configurations to point to your Confluent Cloud broker and schema registry instance, along with the topics you've created in Step 1:
+
+```
+kafka:
+      bootstrap:
+        server: pkc-e8mp5.eu-west-1.aws.confluent.cloud:9092
```
Comment (shirshanka) [inline docs/deploy/confluent-cloud.md:83]: are these real values?

File: docs/deploy/confluent-cloud.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+KAFKA_BOOTSTRAP_SERVER=pkc-e8mp5.eu-west-1.aws.confluent.cloud:9092
+KAFKA_SCHEMAREGISTRY_URL=https://psrc-yorrp.us-east-2.aws.confluent.cloud
+
+# Confluent Cloud Configs
+SPRING_KAFKA_PROPERTIES_SECURITY_PROTOCOL=SASL_SSL
+SPRING_KAFKA_PROPERTIES_SASL_JAAS_CONFIG=org.apache.kafka.common.security.plain.PlainLoginModule   required username='XFA35EL7QFUQP2PA' password='zmkfP6Envr9TYutsjLB3ZYfrk+yfCXD8sQHCE3EMp57A2jNs4RR7J1bU9k6lM6rU';
+SPRING_KAFKA_PROPERTIES_SASL_MECHANISM=PLAIN
+SPRING_KAFKA_PROPERTIES_CLIENT_DNS_LOOKUP=use_all_dns_ips
+SPRING_KAFKA_PROPERTIES_BASIC_AUTH_CREDENTIALS_SOURCE=USER_INFO
+SPRING_KAFKA_PROPERTIES_BASIC_AUTH_USER_INFO=U4UXEN6SP3KEWMO4:ESju8ZfETDm2RZo/6KO057hPYs2TGjFKmvMWUFnlJ3uKubFbB1Sfs7aOjUVT6CGe
```
Comment (shirshanka) [inline docs/deploy/confluent-cloud.md:39]: are these real values?

Comment (jjoyce0510) [reply docs/deploy/confluent-cloud.md]: not anymore

File: docs/deploy/confluent-cloud.md (hunk 2 above)
Comment (jjoyce0510) [reply docs/deploy/confluent-cloud.md]: not anymore

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**